task/write-file/0.1/write-file.yaml

```diff
@@ -0,0 +1,50 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: write-file
+  labels:
+    app.kubernetes.io/version: "0.1"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: tools
+    tekton.dev/displayName: Write a file
+spec:
+  description: >-
+    Write a file to a workspace
+
+    This task can be used to write a file onto the output workspace.
+    Use parameter expansion to insert variable content into the written
+    file. It can also set specific permissions on the file.
+  params:
+    - name: path
+      type: string
+      description: >
+        Relative path to create within the workspace. Directories will
+        be created as necessary.
+    - name: contents
+      type: string
+      description: >
+        Contents of the file to create. Note that octal numbers
+        need quoting in YAML.
+    - name: permissions
```

Comment:
Waking up this morning, I realized that this should perhaps be mode? It seems Ansible, Puppet and Chef all call this parameter mode.

Reply:
Fixed.